nativescript-angular/http-client/ns-http-backend.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from "@angular/core";
+import {
+    HttpRequest, HttpEvent,
+    XhrFactory, HttpResponse,
+    HttpErrorResponse, HttpXhrBackend
+} from "@angular/common/http";
+import { Observable } from "rxjs/Observable";
+
+import { NSFileSystem } from "../file-system/ns-file-system";
+import { isLocalRequest, handleLocalRequest } from "./http-utils";
+
+@Injectable()
+export class NsHttpBackEnd extends HttpXhrBackend {
+    constructor(xhrFactory: XhrFactory, private nsFileSystem: NSFileSystem) {
+        super(xhrFactory);
+    }
+
+    handle(req: HttpRequest<any>): Observable<HttpEvent<any>> {
+        let result: Observable<HttpEvent<any>>;
+
+        if (isLocalRequest(req.url)) {
+            result = this.handleLocalRequest(req.url);
+        } else {
+            result = super.handle(req);
+        }
+
+        return result;
+    }
+
+    private handleLocalRequest(url: string): Observable<HttpEvent<any>> {
```

Comment:
The name of the method is a bit confusing as it matches the name of the imported function.